test/paper-input-container.html

```diff
@@ -104,13 +124,63 @@
       };
     }
 
-    test('label is invisible if value is not null', function() {
-      var nodes = cloneAndAppendTemplate('default');
-      nodes.i.value = 'foobar';
-      nodes.d.updateLabelVisibility(nodes.i.value);
-      assert.ok(!nodes.d._labelVisible);
+    suite('label position', function() {
+
+      test('label is visible by default', function() {
+        var container = fixture('basic');
+        assert.equal(getComputedStyle(container.querySelector('#l')).visibility, 'visible', 'label has visibility:visible');
+      });
+
+      test('label is floated if value is initialized to not null', function() {
+        var container = fixture('has-value');
+        assert.notEqual(getComputedStyle(container.querySelector('#l')).transform, 'none', 'label has transform');
+      });
+
+      test('label is invisible if no-label-float and value is initialized to not null', function() {
+        var container = fixture('no-float-has-value');
+        assert.equal(getComputedStyle(container.querySelector('#l')).visibility, 'hidden', 'label has visibility:hidden');
+      });
+
+    });
+
+    suite('focused styling', function() {
+
+      test('label is colored when input is focused and has value', function() {
+        var container = fixture('has-value');
+        var label = Polymer.dom(container).querySelector('#l');
+        var input = Polymer.dom(container).querySelector('#i');
+        var color = getComputedStyle(label).color;
+        input.focus();
+        assert.equal(document.activeElement, input, 'input is focused');
+        assert.notEqual(getComputedStyle(label).color, color, 'label has different color when input has focus');
+      });
+
+      test('label is not colored when input is focused and has null value', function() {
+        var container = fixture('basic');
+        var label = Polymer.dom(container).querySelector('#l');
+        var input = Polymer.dom(container).querySelector('#i');
+        var color = getComputedStyle(label).color;
+        input.focus();
+        assert.equal(document.activeElement, input, 'input is focused');
+        assert.equal(getComputedStyle(label).color, color, 'label has same color when input has focus and has null value');
+      });
+
+      test('underline is colored when input is focused', function(done) {
+        var container = fixture('basic');
+        var input = Polymer.dom(container).querySelector('#i');
+        var line = Polymer.dom(container.root).querySelector('.focused-line');
+        assert.notEqual(getComputedStyle(line).transform, 'none', 'line is not colored when input is not focused');
+        line.addEventListener('transitionend', function() {
+          assert.equal(getComputedStyle(line).transform, 'none', 'line is colored when input is focused');
+          done();
+        });
+        input.focus();
+        assert.equal(document.activeElement, input, 'input is focused');
+      });
+
     });
 
+/*
```

Comment:
If a test is pending, you can write `xtest` instead of `test` and the suite will not run it (but print out that it is pending). If the test is not needed, can we remove this comment?